src/main/java/com/google/devtools/build/lib/actions/PathStripper.java

```diff
@@ -72,58 +74,110 @@
  */
 public final class PathStripper {
   /**
-   * Support for mapping config parts of exec paths of an action's inputs and outputs.
+   * Creates a new {@link PathMapper} for action implementation logic to use.
    *
-   * <p>The executor should use this to correctly stage an action for execution.
+   * @param stripOutputPaths should this action strip config prefixes?
+   * @param starlarkMnemonic this action's mnemonic if it's a Starlark action, else null
+   * @param outputRoot       the root path where outputs are written (e.g. "bazel-out"). Actions
+   *                         that don't strip outputs can set this to null.
    */
-  public interface ActionStager {
-    /**
-     * Returns the exec path where the executor should stage an action input or output.
-     *
-     * <p>If the action should be config-stripped ({@link PathStripper}), removes "k8-fastbuild"
-     * from paths like "bazel-out/k8-fastbuild/foo/bar".
-     *
-     * <p>Else returns the artifact's original exec path.
-     */
-    default String getMappedExecPathString(ActionInput artifact) {
-      return map(artifact.getExecPath()).getPathString();
-    }
-
-    /** Same as {@link #getMappedExecPathString(ActionInput)} but for a {@link PathFragment}. */
-    PathFragment map(PathFragment execPath);
-
-    /**
-     * Creates a new action stager for executor implementation logic to use.
-     *
-     * @param spawn the action to stage. If {@link Spawn#stripOutputPaths()} is true, paths like
-     *     "bazel-out/k8-fastbuild/bin/foo" are reduced to "bazel-out/bin/foo". Else they're
-     *     unchanged.
-     * @param outputRoot the root path where outputs are written (e.g. "bazel-out")
-     */
-    static ActionStager create(Spawn spawn, PathFragment outputRoot) {
+  public static PathMapper createForAction(
+      boolean stripOutputPaths,
+      @Nullable String starlarkMnemonic,
+      @Nullable PathFragment outputRoot) {
+    if (stripOutputPaths) {
+      Preconditions.checkNotNull(outputRoot);
       Preconditions.checkState(outputRoot.isSingleSegment());
       Preconditions.checkState(!outputRoot.getPathString().contains("\\"));
-      return spawn.stripOutputPaths() ? actionStripper(outputRoot) : NOOP;
     }
+    return stripOutputPaths ? commandStripper(starlarkMnemonic, outputRoot) : PathMapper.NOOP;
+  }
 
-    /** An {@link ActionStager} that doesn't change paths. */
-    ActionStager NOOP = execPath -> execPath;
+  /**
+   * Creates a new {@link PathMapper} for executor implementation logic to use.
+   *
+   * @param spawn      the action to stage. If {@link Spawn#stripOutputPaths()} is true, paths like
+   *                   "bazel-out/k8-fastbuild/bin/foo" are reduced to "bazel-out/bin/foo". Else
+   *                   they're unchanged.
+   * @param outputRoot the root path where outputs are written (e.g. "bazel-out")
+   */
+  public static PathMapper createForExecutor(Spawn spawn, PathFragment outputRoot) {
```

Comment:
@fmeum small proposal. What do you think about naming these methods `createPathMapperForExecutor` and `createPathMapperForAction` for readability? You mentioned in one of your previous comments to Greg's review that there will be further refactoring of PathMapper so maybe that will address the distinction I'm trying to achieve with renaming here?